src/auth.js

```diff
@@ -117,6 +141,98 @@ class Auth {
     return true;
   }
 
+  /**
+   * Creates a refresh token for a user and stores it on the server.
+   *
+   * @private
+   * @param {String} username
+   * @param {Array} groups
+   * @returns {String}
+   */
+  createRefreshToken(username, groups) {
+    const refreshToken = jwt.sign(
+      {username, groups: JSON.stringify(groups)},
+      this.refreshTokenSecret
+    );
+
+    refreshTokens.push(refreshToken);
+    this.refreshTokens.push(refreshToken);
+    return refreshToken;
+  }
+
+  /**
+   * Creates and returns an access token.
+   *
+   * @private
+   * @param {String} username
+   * @param {Array} groups
+   * @returns {String}
+   */
+  createAccessToken(username, groups) {
+    return jwt.sign(
+      {username, groups: JSON.stringify(groups)},
+      this.accessTokenSecret,
+      {expiresIn: '10m'}
+    );
+  }
+
+  /**
+   * Removes a refresh token from the server.
+   *
+   * @private
+   * @param {String} refreshToken
+   */
+  removeRefreshToken(refreshToken) {
+    refreshTokens = refreshTokens.filter(
+      token => token !== refreshToken
+    );
+
+    this.refreshTokens = refreshTokens;
+  }
+
+  /**
+   * Returns the associated user if the access token is valid,
+   * otherwise returns false.
+   * @param {String} accessToken
+   * @returns {Boolean|Object}
+   */
+  validateAccessToken(accessToken) {
+    let returnValue;
```

Comment:
This won't work. `verify` is async when you supply the callback. Use a Promise here.

```js
return new Promise((resolve, reject) => {
    jwt.verify(accessToken, this.accessTokenSecret, (err, user) => {
      if (error) {
        reject(error);
      } else {
        resolve(user);
      }
    });
})
```

Reply:
> This won't work

Well, it *might*. But that's purely luck. You cannot set a variable like this outside of the scope of an async function and then return it because there's no guarantee that it has anything other than `undefined` when you reach the return statement.